### PR TITLE
Scaffolding models from db, generic repository implemented, some sql procedures added

### PR DESCRIPTION
**Pull request by ArtyomKolosov2**

### CODE REVIEW

File: DIMS-Core/DIMS-Core.Common/Helpers/RepositoriesExceptionHelper.cs [hunk 1]
```diff
@@ -0,0 +1,28 @@
+using DIMS_Core.Common.Exceptions;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace DIMS_Core.Common.Helpers
+{
+    public static class RepositoriesExceptionHelper
```
Comment (PushMeNow) [inline DIMS-Core/DIMS-Core.Common/Helpers/RepositoriesExceptionHelper.cs:10]: Hmm... Why repositories only? You can use this helper in other spaces.

File: DIMS-Core/DIMS-Core.DataAccessLayer/Models/DIMSCoreContext.cs [hunk 2]
```diff
@@ -16,11 +18,30 @@ public DIMSCoreContext(DbContextOptions<DIMSCoreContext> options)
...
+        public virtual DbSet<VUserProgress> VUserProgresses { get; set; }
+        public virtual DbSet<VUserTask> VUserTasks { get; set; }
+        public virtual DbSet<VUserTrack> VUserTracks { get; set; }
+
+        protected override void OnConfiguring(DbContextOptionsBuilder optionsBuilder)
+        {
+            if (!optionsBuilder.IsConfigured)
+            {
+#warning To protect potentially sensitive information in your connection string, you should move it out of source code. You can avoid scaffolding the connection string by using the Name= syntax to read it from configuration - see https://go.microsoft.com/fwlink/?linkid=2131148. For more guidance on storing connection strings, see http://go.microsoft.com/fwlink/?LinkId=723263.
+                optionsBuilder.UseSqlServer("Data Source=(localdb)\\ProjectsV13;Initial Catalog=DIMS-Core.Database;Integrated Security=True;Pooling=False;Connect Timeout=30");
```
Comment (PushMeNow) [inline DIMS-Core/DIMS-Core.DataAccessLayer/Models/DIMSCoreContext.cs:37]: Could you remove it and use middleware?

**Review by PushMeNow — CHANGES_REQUESTED** [review]
Hi dude!
I did some comments. Let me know if any questions and get well faster! :)

**Review by dbogdan2000 — APPROVED** [review]

**Review by PushMeNow — APPROVED** [review]
Good! Approve

**Pull request merged**